include/counter.h

```diff
@@ -29,6 +29,40 @@
 extern "C" {
 #endif
 
+/**@defgroup COUNTER_ALARM_FLAGS Flags used in @ref counter_alarm_cfg.
+ * @{ */
+
+/**
+ * @brief Counter alarm absolute value flag.
+ *
+ * Ticks relation to counter value. If set ticks are treated as absolute value,
+ * else it is relative to the counter reading performed during the call.
+ */
+#define COUNTER_ALARM_FLAG_ABSOLUTE BIT(0)
+
+/**
+ * @brief Counter alarm flag enabling detection of setting alarm too late.
+ *
+ * It is required to distinguish between short alarm which was set too late
+ * and alarm which was long, close to counter range. In order to detect that
+ * absolute value that can be set is limited to:
+ * @ref counter_read() + @ref counter_get_max_relative_alarm()
+ */
+#define COUNTER_ALARM_FLAG_DETECT_LATE_SET BIT(1)
+
+/**
+ * @brief Counter alarm flag enabling error reporting on late alarm setting.
+ *
+ * If flag is set then @ref counter_set_channel_alarm returns -ETIME if
+ * requested alarm was set too late. If flag is not set and late detection is
```

Comment:
This flag is to be used only together with `COUNTER_ALARM_FLAG_DETECT_LATE_SET`, without this flag it is ignored, right? I think it would be good to state it clearly at the beginning of the description.
And what about renaming it to `COUNTER_ALARM_FLAG_LATE_ERROR`?

Reply:
done.